### ISSUE TITLE
How to list all apis and create one

### ISSUE THREAD

**Issue by bilalbayasut** [microweber/microweber#404]
Hi, I love microweber, it's a nice cms, however I can't get around in listing the all apis , and create one.

the documentation seems not comprehensive and when i tried to create api like in the documentation this error message shows : 
The api function hello_world is not defined in the allowed functions list

please help , thanks

**Comment by peter-mw** [comment]
Hi,

to list the API function you must login as admin and go to example.com/api/api_index

To create api function: 

Make a module with file called functions.php or make a file called functions.php in the template folder 


## Simple api function

In `functions.php`

put

```
api_expose('hello_world');
function hello_world(){
    return 'hi';
}

```

And when you go to http://example.com/api/hello_world you should see 'hi'

## Api function with GET/POST parameters

```
api_expose('hello_world');
function hello_world($params){
print_r($params);   
}

```

the `$params` array contains the merged $_GET/$_POST php array 

so if you go to
http://example.com/api/hello_world?msg=whats_up_doc

the `$params['msg']` will be of value whats_up_doc




## Anonymous api function
```
api_expose('hello_world', function($params)){
print_r($params);   
});
```

You can use 

`api_expose` to make function available to all
`api_expose_user` to make function available only to logged users
`api_expose_admin` to make function available only to admins 


Hope this helps

**Comment by bilalbayasut** [comment]
thanks I can see json file that looks like listing the existing api, but with there is no documentation yet isn't it ? 

thanks anyway for your reply, really helps

**Comment by bilalbayasut** [comment]
and 1 more question, how can I see the parameters of those api_index ? or where the code is located? because I can't find any documentation of the existing apis

**Comment by peter-mw** [comment]
hi, some of the functions are documented here http://docs.microweber.com/functions/README.md , but the documentation is not finished yet

**Comment by bilalbayasut** [comment]
yes, I still can't find the existing definition of api listed in api_index, such as upload. thus, when i want to upload, I'm confused of what the parameters are and the returned values :) , great cms anyway @peter-mw , thanks

**Comment by peter-mw** [comment]
Hello,
uploading a file is described here http://microweber.com/docs/js-css/mw.uploader.md

**Comment by zahirzohair** [comment]
is there anyone who can help me with microweber API? I mean like for example, how user_register function works? what are the required parameter to pass and what are the return values ? 
any instruction please?
